src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,160 @@
+// @flow
+
+import React from 'react';
+import memoize from 'lodash/memoize';
+import { css } from 'emotion';
+import numeral from 'numeral';
+
+import { Select } from '../Select';
+import { Button } from '../Button';
+import { Icon } from '../Icon';
+import { Text } from '../Text';
+import { theme, PaginationTag, PaginationItemsTag, PaginationItemTag } from './Pagination.theme';
+
+type PaginationProps = {
+  page?: number,
+  pageSize?: number,
+  defaultPage: number,
+  defaultPageSize: number,
+  total: number,
+  showSizeChanger?: boolean,
+  pageSizeOptions?: number[],
+  onChange?: (page: number, pageSize: number) => any,
+};
+
+type PaginationState = {
+  page: number,
+  pageSize: number,
+};
+
+const VISIBLE_RANGE = 2;
+
+class Pagination extends React.Component<PaginationProps, PaginationState> {
+  static defaultProps = {
+    defaultPage: 1,
+    defaultPageSize: 10,
+    total: 0,
+    showSizeChanger: false,
+    pageSizeOptions: [10, 20, 30, 40],
+  };
+
+  constructor(props: PaginationProps) {
+    super(props);
+
+    this.state = {
+      page: props.defaultPage,
+      pageSize: props.defaultPageSize,
+    };
+  }
+
+  // $FlowFixMe
+  onChange = memoize((page) => () => {
+    const { onChange, pageSize: pageSizeFormProps } = this.props;
+
+    this.setState(({ pageSize: pageSizeFromState }) => {
+      const pageSize = pageSizeFormProps || pageSizeFromState;
+
+      if (typeof onChange === 'function') {
+        onChange(page, pageSize);
```

Comment:
Is it fine to call `onChange` inside of `setState` function?

Reply:
@vorobeez  fIxed